### DISCUSSION TITLE
How to install packages into quay.io/strimzi/kafka:0.29.0-kafka-3.2.0 image

### DISCUSSION THREAD

**Discussion by mirisu2** [strimzi/strimzi-kafka-operator#7038] [General]
Hi,
I have to install krb5-workstation krb5-libs packages into quay.io/strimzi/kafka:0.29.0-kafka-3.2.0 image
but
```
[kafka@data-platform-connect-847c6cfd6-kfpmh kafka]$ yum install krb5-workstation krb5-libs
bash: yum: command not found
[kafka@data-platform-connect-847c6cfd6-kfpmh kafka]$ dnf install krb5-workstation krb5-libs
bash: dnf: command not found
```
KafkaConnector has to use Kerberos Authentication. 
What should i do to get the desired result? Can anyone help me?

**Comment by scholzj** [comment]
I think the best way is to write your own Dockerfile where you start from the Strimzi base image, switch the user to `root` (since the default user will not have rights to install anything) and then install the packages using the `micro-dnf` tool => so that should be something like `micro-dnf install ...`. I have no idea what images are available in the repositories, so you have to see whether the Kerberos packages are there.

**Reply by mirisu2** [reply]
> I think the best way is to write your own Dockerfile where you start from the Strimzi base image, switch the user to `root` (since the default user will not have rights to install anything) and then install the packages using the `micro-dnf` tool => so that should be something like `micro-dnf install ...`. I have no idea what images are available in the repositories, so you have to see whether the Kerberos packages are there.

@scholzj Thank you for reply
I was already trying to build custom image
```
FROM quay.io/strimzi/kafka:0.29.0-kafka-3.2.0
USER root:root
RUN yum install -y krb5-workstation krb5-libs
````
and I got an error
```
Sending build context to Docker daemon  65.94MB
Step 1/7 : FROM quay.io/strimzi/kafka:0.29.0-kafka-3.2.0
 ---> e6d7942e869e
Step 2/7 : USER root:root
 ---> Running in 55e4c7c6b42a
Removing intermediate container 55e4c7c6b42a
 ---> e1b64466d8e0
Step 3/7 : RUN yum install -y krb5-workstation krb5-libs
 ---> Running in 413afc902dd2
/bin/sh: yum: command not found
```

**Reply by scholzj** [reply]
Yeah, as I said, you need to use `microdnf` to install the packages, not `yum` (sorry, I mentioned `micro-dnf` in the previous comment, but it is `microdnf` without `-`).

**Comment by mirisu2** [comment]
@scholzj Thank you very much! You save my day

**Reply by scholzj** [reply]
Does it work? My understanding is that Kerberos normally needs to periodical processes to refresh the tokens etc., so not sure how easy it is to actually use it.

**Comment by mirisu2** [comment]
@scholzj Yes. it works. For now I run a command
```
kinit -kt /etc/krb5.keytab userName@domainName -r60d
```
I think I will run another pod or maybe sidecar or job to automatic refresh tokens. There are many solutions.
